### PR TITLE
feat: used tailwind, added todo-list-item and modal view, added wrapper for todo-list

### PR DESCRIPTION
**Pull request by nosensus**
1. Added wrapper for todoList
2. Added Tailwind css and styled ui
3. Added **TodoListItem** component and used it inside **TodoList**.
4. For Edit and Add new todo item I have use the same form, but save to DBFake is not implemented yet. Only ui logic works now.
5. Some errors I have got in components, but everything compile well. Try to pull current branch and make it out. Maybe this problem only on my local machine.

### CODE REVIEW

File: src/components/todo-list-edit-modal/TodoListEditModal.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+type TodoItemProps = {
+  todoItem: TodoItem;
+  isEdit: false;
+  onClose: boolean;
+};
+
+const TodoListEditModal = ({ todoItem, isEdit, onClose }: TodoItemProps) => {
+  if (!isEdit) {
+    todoItem = newTodoItem;
+  }
```
Comment (adizka) [inline src/components/todo-list-edit-modal/TodoListEditModal.tsx]: What is this? You can't change component props like this. Please read/learn the article: https://www.developerway.com/posts/react-re-renders-guide

File: src/components/todo-list-edit-modal/TodoListEditModal.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+              <div className="mb-4">
+                <label className="block text-gray-700 text-sm font-bold mb-2" htmlFor="category">
+                  Category
+                </label>
+                <select
+                  id="category"
+                  name="category"
+                  defaultValue={todoItem.category}
+                  className="shadow border rounded w-full py-2 px-3 text-gray-700 leading-tight"
```
Comment (adizka) [inline src/components/todo-list-edit-modal/TodoListEditModal.tsx]: Where is the `onChnage` attribute?

File: src/components/todo-list/TodoListWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { TodoList } from '.';
+
+const TodoListWrapper = () => {
+  return (
+    <>
+      <TodoList />
+    </>
+  );
+};
```
Comment (adizka) [inline src/components/todo-list/TodoListWrapper.tsx]: ??? No make sense extra component. Why do you need it?

**Review by adizka — CHANGES_REQUESTED** [review]
Let's have a call, you made many conceptual mistakes, I see that there is a lack of understanding of React principles and we should discuss them.

File: src/components/todo-list-edit-modal/TodoListEditModal.tsx (hunk 1 above)
Comment (nosensus) [reply src/components/todo-list-edit-modal/TodoListEditModal.tsx]: Removed

File: src/components/todo-list-edit-modal/TodoListEditModal.tsx (hunk 2 above)
Comment (nosensus) [reply src/components/todo-list-edit-modal/TodoListEditModal.tsx]: I will implement it later. In this PR just popup for new and edit cases.

File: src/components/todo-list/TodoListWrapper.tsx (hunk 3 above)
Comment (nosensus) [reply src/components/todo-list/TodoListWrapper.tsx]: You shown me a Wrapper as a trick to handle error or loading state. And I decided it's a good point. Removed.

**Review by adizka — APPROVED** [review]

**Pull request merged**